### ISSUE TITLE
A Primer on Bayesian Optimization to Optimize Hyperparameters

### ISSUE THREAD

**Issue by collins-a** [section-engineering-education/engineering-education#1926]
### Introduction paragraph (2-3 paragraphs):

In the two previous optimization articles, we carried out basic experiments involving building many models with varied hyperparameters. The experiments of the two previous methods ([grid](https://www.section.io/engineering-education/grid-search/) and [random](https://www.section.io/engineering-education/random-search-hyperparameters/) search) were independent of each other. This independence means that we cannot take information from one experiment to improve another experiment. However, there exists a more intuitive method of hyperparameter optimization. This method uses the outcome of previous iterations to select the next potential hyperparameter values. It is known as Bayesian optimization. This article provides a primer on Bayesian optimization.

Bayesian optimization can be categorized as a sequential model-based optimization algorithm. This means that it carries out trials iteratively as it tries better hyperparameters. This form of optimization uses the outcome of previous iterations to decide on the next hyperparameter values. Unlike grid search and random search, Bayesian optimization does not search through the hyperparameter space blindly. This form of optimization first constructs a probabilistic model. It then maps hyperparameters to a probability of a score on a function known as the objective function.

The objective function is a function that takes in hyperparameters and gives out a score that should be minimized (or maximized). The score produced is a single score and is a real value. Objective functions are costly to compute. It would be possible to attempt every single hyperparameter combination if the objective function could be calculated quickly. As such, Bayesian model-based optimization aims to cut down on the number of times we would need to run the objective function. It calculates the conditional probability of the value of the objective function for a validation set, provided a set of hyperparameter values used to train the probabilistic model. Conditional probability refers to the probability of an event taking place, given that a different event has already taken place. 

### Key takeaways: 

The reader should understand:

1.	Basics of Bayesian Statistics
2.	The theory behind Bayesian optimization
3.	Bayesian optimization application (Steps for use for hyperparameter search)
4.	Surrogate and acquisition functions

### References: 

[Hyperparameter Search: Bayesian Optimization](https://medium.com/analytics-vidhya/hyperparameter-search-bayesian-optimization-14be6fbb0e09)

[Comparison of Hyperparameter Tuning algorithms: Grid search, Random search, Bayesian optimization](https://medium.com/analytics-vidhya/comparison-of-hyperparameter-tuning-algorithms-grid-search-random-search-bayesian-optimization-5326aaef1bd1)

[A Conceptual Explanation of Bayesian Hyperparameter Optimization for Machine Learning](https://towardsdatascience.com/a-conceptual-explanation-of-bayesian-model-based-hyperparameter-optimization-for-machine-learning-b8172278050f#:~:text=%20Bayesian%20Optimization%20%201%20Sequential%20Model-Based%20Optimization.,real-valued%20score%20that...%204%20History.%20%20More%20)

**Comment by hectorkambow** [comment]
very cool - approved 👍  @collins-a
